### ISSUE TITLE
Allow certain object types to occupy only specified layers

### ISSUE THREAD

**Issue by s-m-k** [mapeditor/tiled#1979]
For example, I want to have a separate layer for lights, called "Lights", and a separate layer for enemies, called "Enemies". I can do that already, but it would be a nice further workflow improvement if Tiled prevented me to accidentally place an enemy type onto the "Lights" layer.

The allowed set of layers could be specified either in the object types window or in the object templates.

**Comment by bjorn** [comment]
Yeah, I understand this would convenient. The same functionality would be useful for tiles.

We could allow placing some stuff only in some layers, or we could automatically place some stuff in some layer. I think I'd prefer the latter, cause I think the use-case of having multiple possible layers for something is rare and I think just disallowing is less useful than automatically doing the right thing.

Some issues with specifying this in the object type, is that most objects (except templates and tile objects) are placed without type, and that the type can be changed later. So if implemented like that we need to be make it clear that:

* This feature can only work for tile objects (with the type specified on the tile) or templates (either based on a typed tile, or with a type specified on the template object).
* When you change the type later, or when you change the layer in which that type is supposed to be, existing objects won't be moved.

Of course, other object creation tools could be extended to allow specifying the type of the object they should place, so they too can add the object to the right layer.

Unfortunately, such a change will not be trivial. A lot of code is currently based on adding objects to the "current layer". This includes all tools as well as pasting of objects. It is also the question what it then still means to have a certain layer selected, and whether the current layer should adjust to follow the selected tile / template.
